test/functorch/test_control_flow.py

```diff
@@ -1471,44 +1726,860 @@ def fct_pointwise(x, y):
+
+        def add_scalar_carry(x: torch.Tensor, y: torch.Tensor):
+            return x + 1.0, x + y
+
+        result_init = scan_fct(
+            add_scalar_carry, input=inp, dim=dim, reverse=reverse, init=init
+        )
+        result_exp = _fake_scan(
+            add_scalar_carry, init=init, input=inp, dim=dim, reverse=reverse
+        )
+        self.assertEqual(result_init, result_exp)
+        self.assertEqual(result_init[0], torch.tensor([3.0], device=device))
+
+        # Init with two timestep on dim axis. Should work as y has always 1 on dim axis and
+        # hence automatic broadcasting should work
+        # TODO: However, this creates a problem with the output of scan.
+        # If the carry is allowed to have a scan dim > 1, then the output of the operator will also
+        # be broadcasted to that shape. This means that the output of the operator may not have the same
+        # scan dim as the input, which may be weird.
+        init = torch._ops.ops.aten.slice(x, dim, 0, 2, 1)
+        result_init = scan_fct(op, input=inp, dim=dim, reverse=reverse, init=init)
+        result_exp = _fake_scan(op, init=init, input=inp, dim=dim, reverse=reverse)
+        self.assertEqual(result_init, result_exp)
+        self.assertEqual(result_init[0].shape, torch.Size([2, 5, 2]))
+
+        # Init wrong shape (Other dim different)
+        init = torch._ops.ops.aten.slice(x, dim, 0, 1, 1)
+        init = torch.tile(init, (1, 2, 1))
+        with self.assertRaisesRegex(
+            RuntimeError,
+            "The size of tensor a.*",
+        ):
+            result_init = scan_fct(op, input=inp, dim=dim, reverse=reverse, init=init)
+
+        def add_scalar_carry_sliced_out(x: torch.Tensor, y: torch.Tensor):
+            return x + 1.0, x[:, :1, :] + y
+
+        result_init = scan_fct(
+            add_scalar_carry_sliced_out, input=inp, dim=dim, reverse=reverse, init=init
+        )
+        result_exp = _fake_scan(
+            add_scalar_carry_sliced_out, init=init, input=inp, dim=dim, reverse=reverse
+        )
+        self.assertEqual(result_init, result_exp)
+        self.assertEqual(result_init[0].shape, torch.Size([1, 10, 2]))
+        self.assertEqual(result_init[1].shape, inp.shape)
+
+        # Init wrong pytree
+        init = (
+            torch._ops.ops.aten.slice(x, dim, 0, 1, 1),
+            torch._ops.ops.aten.slice(x, dim, 0, 1, 1),
+        )
+
+        def add_one_carry(x: torch.Tensor, y: torch.Tensor):
+            return x[0], x
+
+        with self.assertRaisesRegex(
+            # Should be: RuntimeError: The number of leaves of the pytree of the new carry produced
+            # by the operator needs to match the length of the pytree of the init
+            torch._dynamo.exc.Unsupported,
+            "Observed exception.*",
+        ):
+            result_init = scan_fct(
+                add_one_carry, input=inp, dim=dim, reverse=reverse, init=init
+            )
```

Comment:
Mind seperating out all raisesRegex tests into differnt tests (with minimal parameterization)? This would make each unit tests standalone. Mind auditing all the places?

Reply:
I separated the raisesRegex out into different test with minimal parameterization as you suggested